src/shogun/base/SGObject.cpp

```diff
@@ -701,6 +712,19 @@ bool CSGObject::has_parameter(const BaseTag& _tag) const
 	return self->has(_tag);
 }
 
+void CSGObject::add_callback_function(
+    const std::string& name, std::function<void()> function)
+{
+	REQUIRE(function, "Function object is not callable");
+	BaseTag tag(name);
+	REQUIRE(
+	    has_parameter(tag), "There is no parameter called \"%s\" in %s\n",
+	    tag.name().c_str(), get_name());
+
+	auto& param = self->at(tag);
+	param.add_callback_function(function);
```

Comment:
shouldn't you move this to avoid copying the callback function? I think function pointer copies are quite expensive..

Reply:
good point